doc/en/components/grids/_shared/row-drag.md

```diff
@@ -0,0 +1,798 @@
+        this.moveRow(targetRow, cursorPosition);
+    }
+
+    private moveRow(draggedRow: RowType, cursorPosition: Point): void {
+        // const parent: IgxHierarchicalGridComponent = (draggedRow as any).grid;
+        // const parent = args.drag.ghostContext.grid;
+        const parent = this.hGrid;
+        const rowIndex: number = this.getTargetRowIndex(parent.rowList.toArray(), cursorPosition);
+        if (rowIndex === -1) { return; }
+        const wasSelected = draggedRow.selected;
+        draggedRow.delete();
+        parent.data.splice(rowIndex, 0, draggedRow.data);
+        if (wasSelected) {
+            parent.selectRows([parent.rowList.toArray()
+                .find((r) => r.rowID === draggedRow.key).rowID], false);
+        }
+    }
+
+    private getTargetRowIndex(rowListArr: RowType[], cursorPosition: Point): number {
+        const targetElem: IgxHierarchicalRowComponent = this.catchCursorPosOnElem(rowListArr, cursorPosition);
+        return rowListArr.indexOf(rowListArr.find((r) => r.rowData.id === targetElem.rowData.id));
+    }
+
+    private catchCursorPosOnElem(rowListArr: any[], cursorPosition: Point)
+        : IgxHierarchicalRowComponent {
+        for (const row of rowListArr) {
+            const rowRect = row.nativeElement.getBoundingClientRect();
+            if (cursorPosition.y > rowRect.top + window.scrollY && cursorPosition.y < rowRect.bottom + window.scrollY &&
+                cursorPosition.x > rowRect.left + window.scrollX && cursorPosition.x < rowRect.right + window.scrollX) {
+                return row;
+            } else if (row === rowListArr[rowListArr.length - 1] && cursorPosition.y > rowRect.bottom) {
+                return row;
+            }
+        }
+    }
+}
+```
+<!-- ComponentEnd: HierarchicalGrid -->
+
+With these few easy steps, you've configured a grid that allows reordering rows via drag/drop! You can see the above code in action in the following demo.
+<!-- ComponentStart: Grid -->
+Holding onto the drag icon will allow you to move a row anywhere in the grid:
+
+<code-view style="height:830px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-row-reorder"
+           github-src="{ComponentSample}/row-reorder"
+           alt="{Platform} {ComponentName} Row Reorder">
+</code-view>
+
+<!-- ComponentEnd: Grid -->
+<!-- ComponentStart: TreeGrid, HierarchicalGrid -->
+Notice that we also have row selection enabled and we preserve the selection when dropping the dragged row.
+
+<code-view style="height:560px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-grid-row-reordering"
+           github-src="{ComponentSample}/row-reorder"
+           alt="{Platform} {ComponentName} Row Reorder">
+</code-view>
+
+<!-- ComponentEnd: TreeGrid, HierarchicalGrid -->
```

Comment:
replace above 3 code-view blocks with just one code-view and remove component build flag because routing path to the sample will be the same for all grid samples in Sample Browser.